store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/handler/AddElementsFromHdfsHandler.java

```diff
@@ -52,9 +79,98 @@ public void doOperation(final AddElementsFromHdfs operation, final AccumuloStore
         }
     }
 
+    private void validateOperation(final AddElementsFromHdfs operation) {
+        if (null != operation.getMinMapTasks()) {
+            LOGGER.warn("minMapTasks field will be ignored");
+        }
+
+        if (null != operation.getMaxMapTasks()) {
+            LOGGER.warn("maxMapTasks field will be ignored");
+        }
+    }
+
+    /**
+     * If there are less then 2 split points and more than 1 tablet server
+     * we should calculate new splits.
+     *
+     * @param store the accumulo store
+     * @return true if the table needs splitting
+     * @throws OperationException if calculating the number of splits or the number of tablet servers fails.
+     */
+    private boolean needsSplitting(final AccumuloStore store) throws OperationException {
+        boolean needsSplitting = false;
+
+        final boolean lessThan2Splits;
+        try {
+            lessThan2Splits = store.getConnection().tableOperations().listSplits(store.getProperties().getTable(), 2).size() < 2;
+        } catch (TableNotFoundException | AccumuloSecurityException | StoreException | AccumuloException e) {
+            throw new OperationException("Unable to get accumulo's split points", e);
+        }
+
+        if (lessThan2Splits) {
+            final int numberTabletServers;
+            try {
+                numberTabletServers = store.getTabletServers().size();
+            } catch (final StoreException e) {
+                throw new OperationException("Unable to get accumulo's tablet servers", e);
+            }
+
+            if (numberTabletServers > 1) {
+                needsSplitting = true;
+            }
+        }
+
+        return needsSplitting;
+    }
+
+    private void sampleAndSplit(final AddElementsFromHdfs operation, final Context context, final AccumuloStore store) throws OperationException {
+        LOGGER.info("Starting to sample input data to create splits points to set on the table");
+
+        // Sample data for split points and split the table
+        final Class<? extends MapperGenerator> mapperGeneratorClass;
+        try {
+            mapperGeneratorClass = Class.forName(operation.getMapperGeneratorClassName()).asSubclass(MapperGenerator.class);
+        } catch (final ClassNotFoundException e) {
+            throw new IllegalArgumentException("Mapper generator class name was invalid: " + operation.getMapperGeneratorClassName(), e);
+        }
+
+        final String outputPath = "/tmp/" + context.getJobId() + "/output";
```

Comment:
Should this be taken from an option on the operation? Putting split points from a table of data which has visibilities to protect the data into /tmp could be unsecure.